tests/python/pants_test/engine/test_isolated_process.py

```diff
@@ -84,22 +154,125 @@ def java_sources_to_javac_args(sources_snapshot, out_dir):
                                    directories_to_create=(out_dir.path,))
 
 
-def process_request_from_java_sources(binary):
-  env = []
-  return ExecuteProcessRequest(
-    argv=binary.prefix_of_command() + ('-version',),
-    env=env,
-    input_files_digest="e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855",
+class JavacVersionCommand(Javac):
+
+  def gen_argv(self, snapshots=None):
```

Comment:
If we remove the abstract method, the snapshots argument can disappear :)

Reply:
Agreed, I'm not sure why I left the `@abstractmethod` in.